src/core/qgspainting.h

```diff
@@ -42,6 +54,49 @@ class CORE_EXPORT QgsPainting
      */
     static bool isClippingMode( Qgis::BlendMode mode );
 
+    /**
+     * Calculates the QTransform which maps the triangle defined by the points (\a inX1, \a inY1), (\a inY2, \a inY2), (\a inX3, \a inY3)
+     * to the triangle defined by (\a outX1, \a outY1), (\a outY2, \a outY2), (\a outX3, \a outY3).
+     *
+     * \param inX1 source triangle vertex 1 x-coordinate
+     * \param inY1 source triangle vertex 1 y-coordinate
+     * \param inX2 source triangle vertex 2 x-coordinate
+     * \param inY2 source triangle vertex 2 y-coordinate
+     * \param inX3 source triangle vertex 3 x-coordinate
+     * \param inY3 source triangle vertex 3 y-coordinate
+     * \param outX1 destination triangle vertex 1 x-coordinate
+     * \param outY1 destination triangle vertex 1 y-coordinate
+     * \param outX2 destination triangle vertex 2 x-coordinate
+     * \param outY2 destination triangle vertex 2 y-coordinate
+     * \param outX3 destination triangle vertex 3 x-coordinate
+     * \param outY3 destination triangle vertex 3 y-coordinate
+     * \param ok will be set to TRUE if the transform could be determined.
+     *
+     * \returns Calculated transform (if possible)
+     *
+     * \since QGIS 3.34
+     */
+    static QTransform triangleToTriangleTransform( double inX1, double inY1, double inX2, double inY2, double inX3, double inY3, double outX1, double outY1, double outX2, double outY2, double outX3, double outY3, bool &ok SIP_OUT );
+
+    /**
+     * Draws a \a triangle onto a \a painter using a mapped texture image.
+     *
+     * The triangle will be rendered using the portion of the texture image described by the triangle (\a textureX1, \a textureY1), (\a textureX2, \a textureY2), (\a textureX3, \a textureY3).
+     * The caller must ensure that \a triangle is a closed QPolygonF consisting of 4 vertices (the 3 triangle vertices + the first vertex again to close the polygon).
+     *
+     * Returns TRUE if the triangle could be rendered, or FALSE if it could not (e.g. when the described points are co-linear).
+     *
+     * \since QGIS 3.34
+     */
+    static bool drawTriangleUsingTexture(
+      QPainter *painter,
+      const QPolygonF &triangle,
+      const QImage &textureImage,
+      double textureX1, double textureY1,
+      double textureX2, double textureY2,
+      double textureX3, double textureY3
+    );
```

Comment:
Note: in GLTF there could be several modes of how the texture coordinates get handled outside of [0-1] range: https://learnopengl.com/img/getting-started/texture_wrapping.png

I guess QBrush only supports the GL_REPEAT mode (can't see a way to change it). But probably the alternative texture wrapping modes are not used much (no idea)...

Reply:
Yeah -- I can't see anyway to reproduce those using standard QPainter methods.